pkg/router/controller/extended_validator.go

```
@@ -40,9 +41,69 @@ func (p *ExtendedValidator) HandleNode(eventType watch.EventType, node *kapi.Nod
+	}
 	return p.plugin.HandleEndpoints(eventType, endpoints)
 }
 
+func validateEndpointAddress(address string) error {
+	ip := net.ParseIP(address)
+	if ip != nil {
+		return checkRestrictedIP(ip)
+	}
+
+	// If not a valid IP, assume it is an FQDN and resolve it
+	ips, err := net.LookupIP(address)
```

golangci-lint (2.12.2)

[error] net.LookupIP must not be called. use (*net.Resolver).LookupIPAddr with a context

(noctx)